internal/services/network/application_gateway_resource.go

```diff
@@ -3541,6 +4109,67 @@ func flattenApplicationGatewayRequestRoutingRules(input *[]network.ApplicationGa
 	return results, nil
 }
 
+func flattenApplicationGatewayRoutingRules(input *[]network.ApplicationGatewayRoutingRule) ([]interface{}, error) {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results, nil
+	}
+
+	for _, config := range *input {
+		if props := config.ApplicationGatewayRoutingRulePropertiesFormat; props != nil {
+			output := map[string]interface{}{
+				"rule_type": string(props.RuleType),
+			}
+
+			if config.ID != nil {
+				output["id"] = *config.ID
```

Comment:
Can we parse and then construct the ID, setting its `.ID()` value back to state here?